docs/ossm/ossm2-migration/gateway-migration.md

```diff
@@ -0,0 +1,92 @@
+# Gateway Migration Guide
+
+Migrating gateways between Istio control planes during a version upgrade from 2.6 to 3.0 is very similar to migrating regular workloads, here is some information on how to migrate your gateways.
+
+## Migration Scenarios
+
+### Gateway Canary Migration (Recommended)
+
+For gradual rollout using multiple gateway versions (avoiding any downtime):
+
+1. Label the gateway namespace to ensure injection from the new mesh is enabled for the namespace (this differs between multitenancy and cluster-wide meshes), ensuring to add the `maistra.io/ignore-namespace: "true"` label as well as remove `istio-injection=enabled` if needed.
+
+2. Deploy a canary gateway (example):
+   ```yaml
+   apiVersion: apps/v1
+   kind: Deployment
+   metadata:
+     name: istio-ingressgateway-canary
+     namespace: istio-ingress # in the same namespace as your existing gateway
+   spec:
+     selector:
+       matchLabels:
+         istio: ingressgateway # must match your existing gateway service selector
+     template:
+       metadata:
+         annotations:
+           inject.istio.io/templates: gateway
+         labels:
+           istio: ingressgateway
+           istio.io/rev: canary # Set to your 3.0 control plane revision
+       spec:
+         containers:
+         - name: istio-proxy
+           image: auto
+   ```
+
+3. Ensure that the new gateway deployment is running with new revision and is handling requests.
+   - Check that pods are running and ready
+   - `oc get pod -o yaml | grep istio.io/rev` to check its running new revision
+   - Test a sample route through the gateway
+
+4. Gradually shift traffic between deployments:
+   ```bash
+   # Increase replicas for new gateway
+   oc scale -n istio-ingress deployment/<new_gateway_deployment> --replicas <new_number_of_replicas>
+   
+   # Decrease replicas for old gateway
+   oc scale -n istio-ingress deployment/<old_gateway_deployment> --replicas <new_number_of_replicas>
+   ```
+
+   Repeat this process, incrementally adjusting replica counts until the new gateway handles all traffic to the gateway Service.
+
+Note that this process is near identical to migrating from SMCP-Defined gateways to gateway injection, if the user migrated previously using [this guide](https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/service_mesh/service-mesh-2-x#ossm-migrating-from-smcp-defined-gateways-to-gateway-injection_gateway-migration), this process should be familiar.
+
+### Dedicated Application Gateway Migration (simple)
```

Comment:
Instead of "simple", maybe we can call this "In Place" to keep it consistent with upstream docs and with the Sail upgrade strategies.

Reply:
Good point, done [7bbca39](https://github.com/openshift-service-mesh/sail-operator/pull/217/commits/7bbca39a365f05d52fa16257827819db6c34214c)